plugins/out_prometheus_remote_write/remote_write_conf.c

```diff
@@ -122,6 +126,37 @@ struct prometheus_remote_write_context *flb_prometheus_remote_write_context_crea
         flb_output_net_default("127.0.0.1", 80, ins);
     }
 
+    /* Check if AWS SigV4 authentication is enabled */
+#ifdef FLB_HAVE_SIGNV4
+#ifdef FLB_HAVE_AWS
+    if (ctx->has_aws_auth) {
+        if (!ctx->aws_service) {
+            flb_plg_error(ins, "aws_auth option requires " FLB_HTTP_AWS_CREDENTIAL_PREFIX
+                          "service to be set");
+            flb_free(ctx);
+            return NULL;
+        }
+
+        ctx->aws_provider = flb_managed_chain_provider_create(
+            ins,
+            config,
+            FLB_HTTP_AWS_CREDENTIAL_PREFIX,
+            NULL,
+            flb_aws_client_generator()
+        );
+        if (!ctx->aws_provider) {
+            flb_plg_error(ins, "failed to create aws credential provider for sigv4 auth");
+            flb_free(ctx);
+            return NULL;
+        }
+
+        /* If managed provider creation succeeds, then region key is present */
+        ctx->aws_region = flb_output_get_property(FLB_HTTP_AWS_CREDENTIAL_PREFIX
```

Comment:
Where does this `FLB_HTTP_AWS_CREDENTIAL_PREFIX` come from?

Reply:
all replaced with FLB_PROMETHEUS_AWS_CREDENTIAL_PREFIX